features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
@geky: 

> If this is garunteed to be thread-safe per the C++03 standard, it may be a better solution over the existing SingletonPtr and we may want consider deprecating the SingletonPtr class in favor of this method. But that is a big if at the moment.

The singleton pointer class is still good to create PODs without wrapping everything as a static, in a function and it is immune to the static initialization fiasco.